MetaMask/metamask-design-system
# Initialize Storybook React

## **Description**

This PR introduces a starter setup for `docs/storybook-react` and configures Tailwind CSS to work with Storybook. We set up the `packages/react-components` as a placeholder for testing components within Storybook and successfully integrated MetaMask's design tokens into the Storybook environment. 

Key changes:
- Created a starter for `docs/storybook-react`.
- Configured Tailwind CSS in Storybook (excluding `tailwind-preset`, which is handled in a separate ticket).
- Added `packages/react-components` as a placeholder to facilitate testing in Storybook.
- Integrated MetaMask design tokens to ensure consistent styling within the Storybook environment.
- Cleaned up most of the Storybook starter files, leaving a few necessary ones for future configurations.

## **Related issues**

Fixes: #729

## **Manual testing steps**

1. Navigate to `docs/storybook-react`.
2. Run `yarn storybook` to start the Storybook environment.
3. Verify that the `packages/react-components` placeholder components are loading correctly in Storybook.
4. Check that the Tailwind utilities are working as expected in the components.
5. Confirm that design tokens are being applied properly to the components.
6. Ensure Storybook is functional with the current configuration and no missing files.

## **Screenshots/Recordings**

### **Before**


N/A

### **After**

https://github.com/user-attachments/assets/43b2cc5e-c02b-420c-93dd-65f12d88a1b5





## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
